client/components/ProjectForm.jsx

```diff
@@ -1,117 +1,380 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import chroma from "chroma-js";
+import Router from "next/router";
+import { addModel } from "../utils/apiWrapper";
 import {
+  Alert,
   Button,
   Col,
   Form,
   FormGroup,
   Input,
   Label,
   Modal,
-  ModalHeader,
-  ModalBody,
-  ModalFooter,
-  Row
+  Row,
+  Tooltip
 } from "reactstrap";
-import sdgs from "../utils/goals";
+import Select from "react-select";
+import UNGoalData from "../utils/goals";
+import countryData from "../utils/countries";
+import groupSizeData from "../utils/groups";
+import levelData from "../utils/levels";
 
 import "../public/styles/projectForm.scss";
 
-export default function ProjectForm() {
-  const [sdg, setSdg] = useState(1);
+export default function ProjectForm(props) {
+  const [projectTitle, setProjectTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [groupSize, setGroupSize] = useState("");
+  const [difficulty, setDifficulty] = useState("");
+  const [country, setCountry] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [error, setError] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [selected_sdgs, setSDGs] = useState([]);
   const [modalIsOpen, setIsOpen] = useState(false);
+  const [typedTitle, setTypedTitle] = useState(false);
+  const [typedDesc, setTypedDesc] = useState(false);
+  const [tooltipOpen, setTooltipOpen] = useState(false);
+  const [isWaiting, setIsWaiting] = useState(false);
+  const [hasPressedSubmit, setPressedSubmit] = useState(false);
+
+  const waitAfterStart = 1000;
+  const waitAfterAPICall = 1500;
+  const minTitleLength = 3;
+  const maxTitleLength = 50;
+  const maxDescLength = 350;
+  const hoverOpacity = 0.1;
+  const selectedOpacity = 0.2;
+  const isCentered = true;
+
+  useEffect(() => {
+    if (props.isModalActivated === true) {
+      toggleModal();
+    }
+  }, [props]);
```

Comment:
Can we add toggleModal as a dependency here?

Reply:
Removed the dependency array altogether